uniffi_macros/src/export/utrait.rs

```diff
@@ -0,0 +1,168 @@
+                    None,
+                );
+                impl_items.push(method);
+                global_items.push(ffi_func);
+                global_items.push(trait_meta);
+            }
+            UniffiTraitDiscriminants::Eq => {
+                let method_eq = quote! {
+                    fn uniffi_trait_eq_eq(&self, other: &#self_ident) -> bool {
+                        use ::std::cmp::PartialEq;
+                        uniffi::deps::static_assertions::assert_impl_all!(#self_ident: PartialEq); // This object has a trait method which requires `PartialEq` be implemented.
+                        PartialEq::eq(self, other)
+                    }
+                };
+                let method_ne = quote! {
+                    fn uniffi_trait_eq_ne(&self, other: &#self_ident) -> bool {
+                        use ::std::cmp::PartialEq;
+                        uniffi::deps::static_assertions::assert_impl_all!(#self_ident: PartialEq); // This object has a trait method which requires `PartialEq` be implemented.
+                        PartialEq::ne(self, other)
+                    }
+                };
+                let (ffi_func_eq, method_meta_eq) =
+                    process_uniffi_trait_method(&method_eq, &self_ident, udl_mode)?;
+                let (ffi_func_ne, method_meta_ne) =
+                    process_uniffi_trait_method(&method_ne, &self_ident, udl_mode)?;
+                // metadata for the trait itself.
+                let discr = UniffiTraitDiscriminants::Eq as u8;
+                let trait_meta = crate::util::create_metadata_items(
+                    "uniffi_trait",
+                    &format!("{}_Eq", self_ident.unraw()),
+                    quote! {
+                        ::uniffi::MetadataBuffer::from_code(::uniffi::metadata::codes::UNIFFI_TRAIT)
+                        .concat_value(#discr)
+                        .concat(#method_meta_eq)
+                        .concat(#method_meta_ne)
+                    },
+                    None,
+                );
+                impl_items.push(method_eq);
+                impl_items.push(method_ne);
+                global_items.push(ffi_func_eq);
+                global_items.push(ffi_func_ne);
+                global_items.push(trait_meta);
+            }
+        }
+    }
+    Ok(quote! {
+        #[doc(hidden)]
+        impl #self_ident {
+            #(#impl_items)*
+        }
+        #(#global_items)*
+    })
+}
+
+fn process_uniffi_trait_method(
+    method: &TokenStream,
+    self_ident: &Ident,
+    udl_mode: bool,
+) -> syn::Result<(TokenStream, TokenStream)> {
+    let item = syn::parse(method.clone().into())?;
+
+    let syn::Item::Fn(item) = item else {
+        unreachable!()
+    };
+
+    let ffi_func = gen_ffi_function(
+        &FnSignature::new_method(self_ident.clone(), item.sig.clone())?,
+        &ExportAttributeArguments::default(),
+        udl_mode,
+    )?;
+    // metadata for the trait itself.
```

Comment:
This is the same as the comments above, but I think it should point out the differences between the two types of metadata.